### PR TITLE
Implement Publish, Save Draft and Delete Post

### PR DESCRIPTION
**Pull request by wisnugro**
This PR creates these endpoints:

- `POST /sites/id/pages`, publish new page.
- `DELETE /sites/id/pages/pageId`, delete page.
- `PUT /sites/id/pages/pageId`, update page.

Postman collection https://www.getpostman.com/collections/3ca97547af8112ce2a48

### CODE REVIEW

**Comment by haydarai** [comment]
@satriowisnugroho how to get csrf token?

**Review by andrewongi — COMMENTED** [review]
do we need to move the page schema into a new file?

**Comment by haydarai** [comment]
@satriowisnugroho @pionize yeah I think that's cleaner. Can you do it @satriowisnugroho?

**Comment by haydarai** [comment]
This will need CRUD for Site, let me working on it

**Comment by wisnugro** [comment]
ok, i'll move the page schema into a new file

File: models/Site.js [hunk 1]
```diff
@@ -30,7 +21,14 @@ const siteSchema = new mongoose.Schema({
   isPrivate: Boolean,
   members: [mongoose.Schema.Types.ObjectId], // It might not be needed in near future
   navigations: [navigationSchema],
-  pages: [pageSchema],
+  pages: [
```
Comment (haydarai) [inline models/Site.js]: Can't we just do `[{ type: mongoose.Schema.Types.ObjectId, ref: 'Page' }]`. Any reason why you want to do it like that?

File: models/Page.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const mongoose = require('mongoose');
+const slug = require('mongoose-slug-generator');
+
+const pageSchema = new mongoose.Schema({
+  siteId: { type: mongoose.Schema.Types.ObjectId, ref: 'Site' },
```
Comment (haydarai) [inline models/Page.js]: Can we rename `siteId` to `site`? So when we need to populate it, we can use `.populate('site')`, the same with `authorId` we can rename it to `author`

File: controllers/page.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    page.content = req.body.content || page.content;
+    page.isPublished = req.body.isPublished || page.isPublished;
+    page.slug = slug(req.body.slug, { lower: true, charmap: '' }) || page.slug;
+
+    page.save((err) => {
+      if (err) {
+        return next(err);
+      }
+
+      res.json({ status: 'success' });
```
Comment (haydarai) [inline controllers/page.js]: We can send updated site instead

File: controllers/page.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ * PUT /sites/id/pages/pageId
+ * Update page
+ */
+exports.putPage = (req, res, next) => {
+  const siteId = req.params.id;
+  const pageId = req.params.pageId;
+
+  Page.findOne({ _id: pageId, siteId }).then((page) => {
+    if (!page) {
+      return res.json({ status: 'Page not found' });
```
Comment (haydarai) [inline controllers/page.js]: Be consistent with this, let's use `message` instead.

**Comment by wisnugro** [comment]
Thanks for your review @haydarai , i'll fix it later.

**Pull request merged**